pkg/ipamd/datastore/data_store.go

```diff
@@ -1538,3 +1546,39 @@ func (ds *DataStore) CheckFreeableENIexists() bool {
 	}
 	return false
 }
+
+// NormalizeCheckpointDataByPodVethExistence will normalize checkpoint data by remove allocations that don't have a corresponding pod veth.
+// This shouldn't happen unless container runtimes have bugs that failed to invoke the cmdDel for died pods.
+// we use this reconciliation as a safety mechanism during transition from CRI to state file.
+func (ds *DataStore) normalizeCheckpointDataByPodVethExistence(checkpoint CheckpointData) (CheckpointData, error) {
+	hostNSLinks, err := ds.netLink.LinkList()
```

Comment:
nit: I am not sure how critical performance is here, but I noticed that we get the list of interfaces here, then we loop over this list for every allocation. If performance is critical, we can pre-process this list and create a hash map that `validateAllocationByPodVethExistence` can use to just hash the `linkNameSuffix` into.

Reply:
@jdn5126 
The performance don't matter here as it's not in critical path such as pod launch(this check only happens during IPAMD restart). Also, the iterations here is only matters of 250 * 250 at most.

I used this instead of the make it a map to avoid depends on the fact that the linkSuffix is always 11 length. We can change this if you have concerns :D

Reply:
Yeah even I was thinking if we need to hash the links and use it for validation but as Yang mentioned this will be run only once during restart and worst case you will have max-pods number of loop.

Reply:
To add some data, with 250 pods, under worse case , the loop take 0.3ms to finish. So i still prefer it than couple the code with a fixed 11 length suffix.

func BenchmarkDummy(b *testing.B) {
	for i := 0; i < b.N; i++ {
		count := 250
		var allocations []string
		var names []string
		for i := 0; i < count; i++ {
			allocations = append(allocations, "dummy1233")
			names = append(names, "abc333333333")
		}
		for _, allocation := range allocations {
			for _, name := range names {
				strings.HasSuffix(name, allocation)
			}
		}
	}
}```